java/vector/src/main/java/org/apache/arrow/vector/BaseFixedWidthVector.java

```diff
@@ -870,12 +870,16 @@ public boolean equals(int index, ValueVector to, int toIndex) {
     if (to == null) {
       return false;
     }
-    if (this.getClass() != to.getClass()) {
+    if (!this.getField().getType().equals(to.getField().getType())) {
       return false;
     }
 
     BaseFixedWidthVector that = (BaseFixedWidthVector) to;
 
+    if (this.isSet(index) != that.isSet(toIndex)) {
+      return false;
+    }
+
     int leftStart = typeWidth * index;
```

Comment:
sorry, I missed this earlier. For null entries, we shouldn't compare the values. The data corresponding to a null entry can be [unspecified](https://github.com/apache/arrow/blob/master/docs/source/format/Layout.rst#example-layout-int32-array).

Reply:
Right, thanks for reminder.